HaveParameter DefaultValue evaluates to '<empty>' when actual value is not empty

### Checklist

- [X] Issue has a meaningful title
- [X] I have searched the existing issues. See [all issues](https://github.com/pester/Pester/issues?q=is%3Aissue)
- [X] I have tested using the latest version of Pester. See [Installation and update guide](https://pester.dev/docs/introduction/installation).

### What is the issue?

Should -HaveParameter -DefaultValue '<anythingOtherThanEmpty>' fails.  Checking for an empty string, when the actual default value is something else, passes instead of failing too.

### Expected Behavior

Should -HaveParameter -DefaultValue should be able to validate default values

### Steps To Reproduce

```powershell
Get-Command -Name Get-ChildItem | Should -HaveParameter -ParameterName 'Path' -DefaultValue '.'
```

### Describe your environment

Pester version     : 5.4.0 C:\Users\brwest\Documents\PowerShell\Modules\Pester\5.4.0\Pester.psm1
PowerShell version : 7.3.1
OS version         : Microsoft Windows NT 10.0.22621.0

### Possible Solution?

Get-ParameterInfo should leverage ASTs since v2 of PowerShell is no longer supported, according to https://pester.dev/docs/introduction/installation#compatibility.  Current implementation requires that param block open parenthesis is on the same line as param keyword for $paramblock to have a count higher than one, which is for the new line, and is still not able to retrieve the default value, unless the parameter name is on the same line as the datatype.

Here's a better example:

```powershell
function Test-Parameter {
    param
    (
        [Parameter()]
        [string]
        $Name = 'test'
    )
}

Get-Command -Name 'Test-Parameter' | Should -HaveParameter -ParameterName 'Name' -DefaultValue 'test'

InvalidResult: Expected command Test-Parameter to have a parameter Name, the default value to be 'test', but the default value was <empty>.
```

I am having difficulty building Pester.  I've installed the 6.0.405 SDK, as well as the .NET Framework 4.5.2 Developer Pack, but I'm getting the following errors, when running .\build.ps1 -Clean from the root of the repository:

  Determining projects to restore...
C:\Users\brwest\git\github\Pester\src\csharp\PesterTests\PesterTests.csproj : error NU1100: Unable to resolve 'Microsoft.NET.Test.Sdk (>= 17.3.2)' for 'net6.0'. [C:\Users\brwest\git\github\Pester\src\csharp\Pest
er.sln]
C:\Users\brwest\git\github\Pester\src\csharp\Pester\Pester.csproj : error NU1100: Unable to resolve 'NETStandard.Library (>= 2.0.3)' for '.NETStandard,Version=v2.0'. [C:\Users\brwest\git\github\Pester\src\csharp
\Pester.sln]
C:\Users\brwest\git\github\Pester\src\csharp\PesterTests\PesterTests.csproj : error NU1100: Unable to resolve 'xunit (>= 2.4.2)' for 'net6.0'. [C:\Users\brwest\git\github\Pester\src\csharp\Pester.sln]
C:\Users\brwest\git\github\Pester\src\csharp\Pester\Pester.csproj : error NU1100: Unable to resolve 'System.Management.Automation (>= 6.0.4)' for '.NETStandard,Version=v2.0'. [C:\Users\brwest\git\github\Pester\s
rc\csharp\Pester.sln]
C:\Users\brwest\git\github\Pester\src\csharp\PesterTests\PesterTests.csproj : error NU1100: Unable to resolve 'xunit.runner.visualstudio (>= 2.4.5)' for 'net6.0'. [C:\Users\brwest\git\github\Pester\src\csharp\Pe
ster.sln]
C:\Users\brwest\git\github\Pester\src\csharp\PesterTests\PesterTests.csproj : error NU1100: Unable to resolve 'coverlet.collector (>= 3.1.2)' for 'net6.0'. [C:\Users\brwest\git\github\Pester\src\csharp\Pester.sl
n]
C:\Users\brwest\git\github\Pester\src\csharp\PesterTests\PesterTests.csproj : error NU1100: Unable to resolve 'System.Management.Automation (>= 6.0.4)' for 'net6.0'. [C:\Users\brwest\git\github\Pester\src\csharp
\Pester.sln]
C:\Users\brwest\git\github\Pester\src\csharp\Pester\Pester.csproj : error NU1100: Unable to resolve 'Microsoft.PowerShell.3.ReferenceAssemblies (>= 1.0.0)' for '.NETFramework,Version=v4.5.2'. [C:\Users\brwest\gi
t\github\Pester\src\csharp\Pester.sln]
C:\Users\brwest\git\github\Pester\src\csharp\Pester\Pester.csproj : error NU1100: Unable to resolve 'Microsoft.NETFramework.ReferenceAssemblies (>= 1.0.0)' for '.NETFramework,Version=v4.5.2'. [C:\Users\brwest\gi
t\github\Pester\src\csharp\Pester.sln]
  Failed to restore C:\Users\brwest\git\github\Pester\src\csharp\PesterTests\PesterTests.csproj (in 114 ms).
  Failed to restore C:\Users\brwest\git\github\Pester\src\csharp\Pester\Pester.csproj (in 114 ms).
MSBuild version 17.4.1+9a89d02ff for .NET
C:\Users\brwest\git\github\Pester\src\csharp\PesterTests\PesterTests.csproj : error NU1100: Unable to resolve 'Microsoft.NET.Test.Sdk (>= 17.3.2)' for 'net6.0'.
C:\Users\brwest\git\github\Pester\src\csharp\PesterTests\PesterTests.csproj : error NU1100: Unable to resolve 'xunit (>= 2.4.2)' for 'net6.0'.
C:\Users\brwest\git\github\Pester\src\csharp\PesterTests\PesterTests.csproj : error NU1100: Unable to resolve 'xunit.runner.visualstudio (>= 2.4.5)' for 'net6.0'.
C:\Users\brwest\git\github\Pester\src\csharp\PesterTests\PesterTests.csproj : error NU1100: Unable to resolve 'coverlet.collector (>= 3.1.2)' for 'net6.0'.
C:\Users\brwest\git\github\Pester\src\csharp\PesterTests\PesterTests.csproj : error NU1100: Unable to resolve 'System.Management.Automation (>= 6.0.4)' for 'net6.0'.
C:\Users\brwest\git\github\Pester\src\csharp\Pester\Pester.csproj : error NU1100: Unable to resolve 'NETStandard.Library (>= 2.0.3)' for '.NETStandard,Version=v2.0'. [TargetFramework=net452]
C:\Users\brwest\git\github\Pester\src\csharp\Pester\Pester.csproj : error NU1100: Unable to resolve 'System.Management.Automation (>= 6.0.4)' for '.NETStandard,Version=v2.0'. [TargetFramework=net452]
C:\Users\brwest\git\github\Pester\src\csharp\Pester\Pester.csproj : error NU1100: Unable to resolve 'Microsoft.PowerShell.3.ReferenceAssemblies (>= 1.0.0)' for '.NETFramework,Version=v4.5.2'. [TargetFramework=ne
t452]
C:\Users\brwest\git\github\Pester\src\csharp\Pester\Pester.csproj : error NU1100: Unable to resolve 'Microsoft.NETFramework.ReferenceAssemblies (>= 1.0.0)' for '.NETFramework,Version=v4.5.2'. [TargetFramework=ne
t452]
C:\Users\brwest\git\github\Pester\src\csharp\Pester\Pester.csproj : error NU1100: Unable to resolve 'NETStandard.Library (>= 2.0.3)' for '.NETStandard,Version=v2.0'. [TargetFramework=netstandard2.0]
C:\Users\brwest\git\github\Pester\src\csharp\Pester\Pester.csproj : error NU1100: Unable to resolve 'System.Management.Automation (>= 6.0.4)' for '.NETStandard,Version=v2.0'. [TargetFramework=netstandard2.0]
C:\Users\brwest\git\github\Pester\src\csharp\Pester\Pester.csproj : error NU1100: Unable to resolve 'Microsoft.PowerShell.3.ReferenceAssemblies (>= 1.0.0)' for '.NETFramework,Version=v4.5.2'. [TargetFramework=ne
tstandard2.0]
C:\Users\brwest\git\github\Pester\src\csharp\Pester\Pester.csproj : error NU1100: Unable to resolve 'Microsoft.NETFramework.ReferenceAssemblies (>= 1.0.0)' for '.NETFramework,Version=v4.5.2'. [TargetFramework=ne
tstandard2.0]

Build FAILED.

C:\Users\brwest\git\github\Pester\src\csharp\PesterTests\PesterTests.csproj : error NU1100: Unable to resolve 'Microsoft.NET.Test.Sdk (>= 17.3.2)' for 'net6.0'.
C:\Users\brwest\git\github\Pester\src\csharp\PesterTests\PesterTests.csproj : error NU1100: Unable to resolve 'xunit (>= 2.4.2)' for 'net6.0'.
C:\Users\brwest\git\github\Pester\src\csharp\PesterTests\PesterTests.csproj : error NU1100: Unable to resolve 'xunit.runner.visualstudio (>= 2.4.5)' for 'net6.0'.
C:\Users\brwest\git\github\Pester\src\csharp\PesterTests\PesterTests.csproj : error NU1100: Unable to resolve 'coverlet.collector (>= 3.1.2)' for 'net6.0'.
C:\Users\brwest\git\github\Pester\src\csharp\PesterTests\PesterTests.csproj : error NU1100: Unable to resolve 'System.Management.Automation (>= 6.0.4)' for 'net6.0'.
C:\Users\brwest\git\github\Pester\src\csharp\Pester\Pester.csproj : error NU1100: Unable to resolve 'NETStandard.Library (>= 2.0.3)' for '.NETStandard,Version=v2.0'. [TargetFramework=net452]
C:\Users\brwest\git\github\Pester\src\csharp\Pester\Pester.csproj : error NU1100: Unable to resolve 'System.Management.Automation (>= 6.0.4)' for '.NETStandard,Version=v2.0'. [TargetFramework=net452]
C:\Users\brwest\git\github\Pester\src\csharp\Pester\Pester.csproj : error NU1100: Unable to resolve 'Microsoft.PowerShell.3.ReferenceAssemblies (>= 1.0.0)' for '.NETFramework,Version=v4.5.2'. [TargetFramework=ne
t452]
C:\Users\brwest\git\github\Pester\src\csharp\Pester\Pester.csproj : error NU1100: Unable to resolve 'Microsoft.NETFramework.ReferenceAssemblies (>= 1.0.0)' for '.NETFramework,Version=v4.5.2'. [TargetFramework=ne
t452]
C:\Users\brwest\git\github\Pester\src\csharp\Pester\Pester.csproj : error NU1100: Unable to resolve 'NETStandard.Library (>= 2.0.3)' for '.NETStandard,Version=v2.0'. [TargetFramework=netstandard2.0]
C:\Users\brwest\git\github\Pester\src\csharp\Pester\Pester.csproj : error NU1100: Unable to resolve 'System.Management.Automation (>= 6.0.4)' for '.NETStandard,Version=v2.0'. [TargetFramework=netstandard2.0]
C:\Users\brwest\git\github\Pester\src\csharp\Pester\Pester.csproj : error NU1100: Unable to resolve 'Microsoft.PowerShell.3.ReferenceAssemblies (>= 1.0.0)' for '.NETFramework,Version=v4.5.2'. [TargetFramework=ne
tstandard2.0]
C:\Users\brwest\git\github\Pester\src\csharp\Pester\Pester.csproj : error NU1100: Unable to resolve 'Microsoft.NETFramework.ReferenceAssemblies (>= 1.0.0)' for '.NETFramework,Version=v4.5.2'. [TargetFramework=ne
tstandard2.0]
    0 Warning(s)
    13 Error(s)

Time Elapsed 00:00:00.55
Exception: C:\Users\brwest\git\github\Pester\build.ps1:83
Line |
  83 |          throw "build failed!"
     |          ~~~~~~~~~~~~~~~~~~~~~
     | build failed!
     
Here's the output from running winget list --name .net:

Microsoft .NET SDK 6.0.405 (x64)                          Microsoft.DotNet.SDK.6                 6.0.405   winget
Microsoft .NET Framework 4.5.2 Multi-Targeting Pack       {19E8AE59-4D4A-3534-B567-6CC08FA4102E} 4.5.51651
Microsoft .NET Framework 4.5.2 Multi-Targeting Pack (ENU) {290FC320-2F5A-329E-8840-C4193BD7A9EE} 4.5.51209
Microsoft .NET SDK 7.0.102 (x64)                          Microsoft.DotNet.SDK.7                 7.0.102   winget
Microsoft .NET Framework 4.8.1 Targeting Pack             {94DDB521-CDD4-4A83-BBE0-D3C856FE9420} 4.8.09032
Microsoft .NET Runtime - 6.0.13 (x64)                     Microsoft.DotNet.Runtime.6             6.0.13    winget
Microsoft .NET Framework 4.8.1 SDK                        {BD4C49AC-2A45-48B0-B3F7-0C6043987AD0} 4.8.09032
Microsoft Windows Desktop Runtime - 7.0.2 (x64)           Microsoft.DotNet.DesktopRuntime.7      7.0.2     winget

I've tried running without the .NET 7 SDK since neither https://github.com/pester/Pester/blob/main/CONTRIBUTING.md#required-software or 'src\csharp\Pester\Pester.csproj' mention it, but that results in:

Found .NET SDK, but did not find dotnet.dll at [C:\Program Files\dotnet\sdk\7.0.102\dotnet.dll]
Found .NET SDK, but did not find dotnet.dll at [C:\Program Files\dotnet\sdk\7.0.102\dotnet.dll]
Exception: C:\Users\brwest\git\github\Pester\build.ps1:83
Line |
  83 |          throw "build failed!"
     |          ~~~~~~~~~~~~~~~~~~~~~
     | build failed!
     
I assume I'm doing something wrong, but since I'm not a C# developer, I'm not sure what it is.  Any help would be greatly appreciated.

--- a/src/functions/assertions/HaveParameter.ps1
+++ b/src/functions/assertions/HaveParameter.ps1
@@ -31,79 +31,42 @@
     }
 
     function Get-ParameterInfo {
-        param(
-            [Parameter( Mandatory = $true )]
+        param (
+            [Parameter(Mandatory = $true)]
             [Management.Automation.CommandInfo]$Command
         )
         <#
         .SYNOPSIS
-            Use Tokenize to get information about the parameter block of a command
+            Use AST to get information about the parameter block of a command
         .DESCRIPTION
             In order to get information about the parameter block of a command,
             several tools can be used (Get-Command, AST, etc).
             In order to get the default value of a parameter, AST is the easiest
-            way to go; but AST was only introduced with PSv3.
-            This function creates an object with information about parameters
-            using the Tokenize
+            way to go
         .NOTES
-            Author: Chris Dent
+            Author: Brian West
         #>
 
-        function Get-TokenGroup {
-            param(
-                [Parameter( Mandatory = $true )]
-                [System.Management.Automation.PSToken[]]$tokens
-            )
-            $i = $j = 0
-            do {
-                $token = $tokens[$i]
-                if ($token.Type -eq 'GroupStart') {
-                    $j++
-                }
-                if ($token.Type -eq 'GroupEnd') {
-                    $j--
-                }
-                if (-not $token.PSObject.Properties.Item('Depth')) {
-                    $token.PSObject.Properties.Add([Pester.Factory]::CreateNoteProperty("Depth", $j))
-                }
-                $token
-
-                $i++
-            } until ($j -eq 0 -or $i -ge $tokens.Count)
-        }
-
-        $errors = $null
-        $tokens = [System.Management.Automation.PSParser]::Tokenize($Command.Definition, [Ref]$errors)
-
         # Find param block
-        $start = $tokens.IndexOf(($tokens | & $SafeCommands['Where-Object'] { $_.Content -eq 'param' } | & $SafeCommands['Select-Object'] -First 1)) + 1
-        $paramBlock = Get-TokenGroup $tokens[$start..($tokens.Count - 1)]
+        $ast = (Get-Command -Name $Command).ScriptBlock.Ast
 
-        for ($i = 0; $i -lt $paramBlock.Count; $i++) {
-            $token = $paramBlock[$i]
+        $paramBlock = $ast.FindAll(
+            {
+                param($Item)
+                return ($Item -is [System.Management.Automation.Language.ParamBlockAst])
+            },
+            $true
+        )
 
-            if ($token.Depth -eq 1 -and $token.Type -eq 'Variable') {
-                $paramInfo = & $SafeCommands['New-Object'] PSObject -Property @{
-                    Name = $token.Content
-                } | & $SafeCommands['Select-Object'] Name, Type, DefaultValue, DefaultValueType
+        foreach ($parameter in $paramBlock.Parameters) {
+            $paramInfo = & $SafeCommands['New-Object'] PSObject -Property @{
+                Name = $parameter.Name.VariablePath.UserPath
+                DefaultValue = $parameter.DefaultValue.Value
+                DefaultValueType = $parameter.StaticType.Name
+                Type = "[$($parameter.StaticType.Name.ToLower())]"
+            } | & $SafeCommands['Select-Object'] Name, Type, DefaultValue, DefaultValueType
 
-                if ($paramBlock[$i + 1].Content -ne ',') {
-                    $value = $paramBlock[$i + 2]
-                    if ($value.Type -eq 'GroupStart') {
-                        $tokenGroup = Get-TokenGroup $paramBlock[($i + 2)..($paramBlock.Count - 1)]
-                        $paramInfo.DefaultValue = [String]::Join('', ($tokenGroup | & $SafeCommands['ForEach-Object'] { $_.Content }))
-                        $paramInfo.DefaultValueType = 'Expression'
-                    }
-                    else {
-                        $paramInfo.DefaultValue = $value.Content
-                        $paramInfo.DefaultValueType = $value.Type
-                    }
-                }
-                if ($paramBlock[$i - 1].Type -eq 'Type') {
-                    $paramInfo.Type = $paramBlock[$i - 1].Content
-                }
-                $paramInfo
-            }
+            $paramInfo
         }
     }
 
@@ -235,7 +198,12 @@
 
         if ($PSBoundParameters.Keys -contains "DefaultValue") {
             $parameterMetadata = Get-ParameterInfo $ActualValue | & $SafeCommands['Where-Object'] { $_.Name -eq $ParameterName }
-            $actualDefault = if ($parameterMetadata.DefaultValue) { $parameterMetadata.DefaultValue } else { "" }
+            $actualDefault = if ($parameterMetadata.DefaultValue) {
+                $parameterMetadata.DefaultValue
+            }
+            else {
+                ""
+            }
             $testDefault = ($actualDefault -eq $DefaultValue)
             $filters += "the default value$(if ($Negate) {" not"}) to be $(Format-Nicely $DefaultValue)"
 
@@ -276,10 +244,10 @@
                     $faultyAliases += $AliasValue
                 }
             }
-            if($faultyAliases.Count -ge 1) {
+            if ($faultyAliases.Count -ge 1) {
                 $aliases = $(Join-And ($faultyAliases -replace '^|$', "'"))
                 $singular = $faultyAliases.Count -eq 1
-                if($Negate) {
+                if ($Negate) {
                     $buts += "it has $(if($singular) {"an alias"} else {"the aliases"} ) $aliases"
                 }
                 else {

--- a/src/functions/assertions/HaveParameter.ps1
+++ b/src/functions/assertions/HaveParameter.ps1
@@ -60,11 +60,19 @@
 
         foreach ($parameter in $paramBlock.Parameters) {
             $paramInfo = & $SafeCommands['New-Object'] PSObject -Property @{
-                Name = $parameter.Name.VariablePath.UserPath
-                DefaultValue = $parameter.DefaultValue.Value
+                Name             = $parameter.Name.VariablePath.UserPath
                 DefaultValueType = $parameter.StaticType.Name
-                Type = "[$($parameter.StaticType.Name.ToLower())]"
+                Type             = "[$($parameter.StaticType.Name.ToLower())]"
             } | & $SafeCommands['Select-Object'] Name, Type, DefaultValue, DefaultValueType
+
+            if ($null -ne $parameter.DefaultValue) {
+                if ([bool](Get-Member -Name Value -InputObject $parameter.DefaultValue -MemberType Property)) {
+                    $paramInfo.DefaultValue = $parameter.DefaultValue.Value
+                }
+                else {
+                    $paramInfo.DefaultValue = $parameter.DefaultValue.Extent.Text
+                }
+            }
 
             $paramInfo
         }

--- a/src/functions/assertions/HaveParameter.ps1
+++ b/src/functions/assertions/HaveParameter.ps1
@@ -47,19 +47,21 @@
             Author: Brian West
         #>
 
-        # Find param block
+        # Find parameters
         $ast = $Command.ScriptBlock.Ast
 
         if ($null -ne $ast) {
-            $paramBlock = $ast.FindAll(
-                {
-                    param($Item)
-                    return ($Item -is [System.Management.Automation.Language.ParamBlockAst])
-                },
-                $true
-            )
+            if ($null -ne $ast.Parameters) {
+                $parameters = $ast.Parameters
+            }
+            elseif ($null -ne $ast.Body.ParamBlock) {
+                $parameters = $ast.Body.ParamBlock.Parameters
+            }
+            else {
+                return
+            }
 
-            foreach ($parameter in $paramBlock.Parameters) {
+            foreach ($parameter in $parameters) {
                 $paramInfo = & $SafeCommands['New-Object'] PSObject -Property @{
                     Name             = $parameter.Name.VariablePath.UserPath
                     DefaultValueType = $parameter.StaticType.Name

--- a/src/functions/assertions/HaveParameter.ps1
+++ b/src/functions/assertions/HaveParameter.ps1
@@ -48,33 +48,35 @@
         #>
 
         # Find param block
-        $ast = (& $SafeCommands['Get-Command'] -Name $Command).ScriptBlock.Ast
+        $ast = $Command.ScriptBlock.Ast
 
-        $paramBlock = $ast.FindAll(
-            {
-                param($Item)
-                return ($Item -is [System.Management.Automation.Language.ParamBlockAst])
-            },
-            $true
-        )
+        if ($null -ne $ast) {
+            $paramBlock = $ast.FindAll(
+                {
+                    param($Item)
+                    return ($Item -is [System.Management.Automation.Language.ParamBlockAst])
+                },
+                $true
+            )
 
-        foreach ($parameter in $paramBlock.Parameters) {
-            $paramInfo = & $SafeCommands['New-Object'] PSObject -Property @{
-                Name             = $parameter.Name.VariablePath.UserPath
-                DefaultValueType = $parameter.StaticType.Name
-                Type             = "[$($parameter.StaticType.Name.ToLower())]"
-            } | & $SafeCommands['Select-Object'] Name, Type, DefaultValue, DefaultValueType
+            foreach ($parameter in $paramBlock.Parameters) {
+                $paramInfo = & $SafeCommands['New-Object'] PSObject -Property @{
+                    Name             = $parameter.Name.VariablePath.UserPath
+                    DefaultValueType = $parameter.StaticType.Name
+                    Type             = "[$($parameter.StaticType.Name.ToLower())]"
+                } | & $SafeCommands['Select-Object'] Name, Type, DefaultValue, DefaultValueType
 
-            if ($null -ne $parameter.DefaultValue) {
-                if ([bool](Get-Member -Name Value -InputObject $parameter.DefaultValue -MemberType Property)) {
-                    $paramInfo.DefaultValue = $parameter.DefaultValue.Value
+                if ($null -ne $parameter.DefaultValue) {
+                    if ($parameter.DefaultValue.PSObject.Properties['Value']) {
+                        $paramInfo.DefaultValue = $parameter.DefaultValue.Value
+                    }
+                    else {
+                        $paramInfo.DefaultValue = $parameter.DefaultValue.Extent.Text
+                    }
                 }
-                else {
-                    $paramInfo.DefaultValue = $parameter.DefaultValue.Extent.Text
-                }
+
+                $paramInfo
             }
-
-            $paramInfo
         }
     }
 

--- a/src/functions/assertions/HaveParameter.ps1
+++ b/src/functions/assertions/HaveParameter.ps1
@@ -48,7 +48,7 @@
         #>
 
         # Find param block
-        $ast = (Get-Command -Name $Command).ScriptBlock.Ast
+        $ast = (& $SafeCommands['Get-Command'] -Name $Command).ScriptBlock.Ast
 
         $paramBlock = $ast.FindAll(
             {

--- a/tst/functions/assertions/HaveParameter.Tests.ps1
+++ b/tst/functions/assertions/HaveParameter.Tests.ps1
@@ -61,15 +61,13 @@ InPesterModuleScope {
         }
         else {
             function Invoke-DummyFunction {
-                param
-                (
+                param(
                     [Parameter(Mandatory = $true)]
                     [Alias('First', 'Another')]
                     $MandatoryParam,
 
                     [ValidateNotNullOrEmpty()]
-                    [DateTime]
-                    $ParamWithNotNullOrEmptyValidation = (Get-Date),
+                    [DateTime]$ParamWithNotNullOrEmptyValidation = (Get-Date),
 
                     # argument completer is PowerShell v5+ only
                     [Parameter()]
@@ -154,6 +152,17 @@ InPesterModuleScope {
             }
         ) {
             Get-Command "Invoke-DummyFunction" | Should -HaveParameter $ParameterName -DefaultValue $ExpectedValue
+        }
+
+        It "passes if the paramblock has opening parenthesis on new line and parameter has a default value" {
+            function Test-Paramblock {
+                param
+                (
+                    $Name = 'test'
+                )
+            }
+
+            Get-Command -Name 'Test-Paramblock' | Should -HaveParameter -ParameterName 'Name' -DefaultValue 'test'
         }
 
         It "passes if the parameter <ParameterName> exists, is of type <ExpectedType> and has a default value '<ExpectedValue>'" -TestCases @(

--- a/tst/functions/assertions/HaveParameter.Tests.ps1
+++ b/tst/functions/assertions/HaveParameter.Tests.ps1
@@ -61,13 +61,15 @@ InPesterModuleScope {
         }
         else {
             function Invoke-DummyFunction {
-                param(
+                param
+                (
                     [Parameter(Mandatory = $true)]
                     [Alias('First', 'Another')]
                     $MandatoryParam,
 
                     [ValidateNotNullOrEmpty()]
-                    [DateTime]$ParamWithNotNullOrEmptyValidation = (Get-Date),
+                    [DateTime]
+                    $ParamWithNotNullOrEmptyValidation = (Get-Date),
 
                     # argument completer is PowerShell v5+ only
                     [Parameter()]

